src/python/pants/init/engine_initializer.py

```diff
@@ -81,8 +82,18 @@ def __init__(self, invalid_goals):
             )
             self.invalid_goals = invalid_goals
 
+    def goal_consumed_subsystem_scopes(self, goal_name: str) -> Tuple[str, ...]:
```

Comment:
If this is the only method we end up using, you could probably make `goal_consumed_types` private, or inline this.